File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    Set = set()
    ls = [0 for _ in range(n)]
    for l in range(n):
        for idx in range(n):
            ls[idx] += elements[(idx + l) % n]
            Set.add(ls[idx])
    
    return len(list(Set))

File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    let answer = new Set();
    let n = elements.length;
    let ls = Array.from({length: n}).fill(0);
    
    for(let i = 0;i <n;i++) {
        for(let idx = 0; idx < n; idx++) {
            ls[idx] += elements[(idx + i) % n];
            answer.add(ls[idx]);
        };
    };
    
    return answer.size;
}